renderlib/Manipulator.h

```diff
@@ -0,0 +1,225 @@
+#pragma once
+
+#include "AppScene.h"
+#include "MathUtil.h"
+#include "RenderSettings.h"
+#include "gesture/gesture.h"
+
+#include <assert.h>
+#include <inttypes.h>
+#include <vector>
+
+struct ManipulationTool
```

Comment:
Question: When is it good practice to use classes vs. structs?

Reply:
This is how I think about it: because structs have all members public by default, they are much simpler for small "plain old data" POD data types.  I would generally use structs for that and classes for anything else that is likely to have methods and need to possibly have public/private/protected stuff. If I am making a struct and need something to be private, I would pause and evaluate my choice.   In this case I can't really think of a good reason why a ManipulationTool should prefer struct over class. I actually think of it more as a class.  But in C++ the distinction is pretty minor.